prov/verbs/src/verbs_domain.c

```diff
@@ -325,19 +325,23 @@ vrb_domain(struct fid_fabric *fabric, struct fi_info *info,
 	struct vrb_fabric *fab =
 		 container_of(fabric, struct vrb_fabric,
 			      util_fabric.fabric_fid);
-	const struct fi_info *fi = vrb_get_verbs_info(vrb_util_prov.info,
-							 info->domain_attr->name);
-	if (!fi)
-		return -FI_EINVAL;
+	struct fi_info *fi = vrb_get_verbs_info(vrb_util_prov.info,
+						info->domain_attr->name);
+	if (!fi) {
+		ret = -FI_EINVAL;
+		goto err0;
```

Comment:
If fi == null, we can just return an error here, rather than call fi_freeinfo() on the null value.